### PR TITLE
Update schemas for breaking COB change

### PR DESCRIPTION
**Pull request by rudolfs**
Update interface to work with the [COB breaking change](https://app.radicle.xyz/nodes/seed.radicle.xyz/rad:z3gqcJUoA1n9HaHKufZs5FCSGazv5/patches/68cdaf75230ca1763cf2baef7397c87663719ba8).
Deploy this only after seed.radicle.xyz has been updated to [3bd5589](https://app.radicle.xyz/nodes/seed.radicle.xyz/rad:z3gqcJUoA1n9HaHKufZs5FCSGazv5/commits/3bd5589b1a7484c2b72ce98da4054b917c123bf0) or later.

### CODE REVIEW

File: src/views/projects/Cob/AssigneeInput.svelte [hunk 1]
```diff
@@ -22,10 +22,11 @@
 
   function addAssignee() {
     if (parsedNodeId) {
-      if (updatedAssignees.includes(parsedNodeId.pubkey)) {
+      const assignee = `${parsedNodeId.prefix}${parsedNodeId.pubkey}`;
+      if (updatedAssignees.includes(assignee)) {
         validationMessage = "This assignee is already added";
       } else {
-        updatedAssignees = [...updatedAssignees, parsedNodeId.pubkey];
+        updatedAssignees = [...updatedAssignees, assignee];
```
Comment (cloudhead) [inline src/views/projects/Cob/AssigneeInput.svelte:29]: This should be did-formatted

File: httpd-client/lib/project/patch.ts [hunk 2]
```diff
@@ -113,37 +113,97 @@ export const patchSchema = object({
...
 
+export type LifecycleState =
+  | { status: "draft" }
+  | { status: "open" }
+  | { status: "archived" };
+
+export type CodeLocation = {
+  path: string;
+  old?: { start: number; end: number };
+  new?: { start: number; end: number };
```
Comment (cloudhead) [inline httpd-client/lib/project/patch.ts:132]: This has changed actually. I'll update the ts schema.

```rust
/// Code range.
#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase", tag = "type")]
pub enum CodeRange {
    /// One or more lines.
    Lines { range: Range<usize> },
    /// Character range within a line.
    Chars { line: usize, range: Range<usize> },
}

/// Code location, used for attaching comments to diffs.
#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CodeLocation {
    /// Path of file.
    pub path: PathBuf,
    /// Line range on old file. `None` for added files.
    pub old: Option<CodeRange>,
    /// Line range on new file. `None` for deleted files.
    pub new: Option<CodeRange>,
}
```

File: src/views/projects/Cob/AssigneeInput.svelte (hunk 1 above)
Comment (rudolfs) [reply src/views/projects/Cob/AssigneeInput.svelte:29]: It is already.

Comment (rudolfs) [reply src/views/projects/Cob/AssigneeInput.svelte:29]: It is. We run it through the `parseNodeId` function above to make sure it's a valid node id and then put it back together on line 25.

File: httpd-client/lib/project/patch.ts (hunk 2 above)
Comment (rudolfs) [reply httpd-client/lib/project/patch.ts:132]: b2fc41e6a23e5d08037fa931e0248fe2b1a1f691

Comment (rudolfs) [reply httpd-client/lib/project/patch.ts]: b2fc41e6a23e5d08037fa931e0248fe2b1a1f691

**Review by cloudhead — APPROVED** [review]

**Pull request merged**